File: app/src/main/java/com/example/storeapp/ViewModels/MainAdapter.java
package com.example.storeapp.ViewModels;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentStateManagerControl;
import androidx.fragment.app.FragmentStatePagerAdapter;
import androidx.lifecycle.Lifecycle;
import androidx.viewpager2.adapter.FragmentStateAdapter;

import com.example.storeapp.Activities.MainActivity;
import com.example.storeapp.Fragments.OrderFragment;
import com.example.storeapp.Fragments.CustomerFragment;
import com.example.storeapp.Fragments.HomePageFragment;

import java.util.List;

public class MainAdapter extends FragmentStateAdapter {
    private List<Fragment> fragmentList;

    public MainAdapter(List<Fragment> fragmentList, FragmentManager fragmentManager, Lifecycle lifecycle) {
        super(fragmentManager, lifecycle);
        this.fragmentList = fragmentList;

    }
    @NonNull
    @Override
    public Fragment createFragment(int position) {
        return fragmentList.get(position);
    }

    @Override
    public int getItemCount() {
        return fragmentList.size();
    }
}

File: app/src/main/java/com/example/myapplication/ViewModels/MainAdapter.kt
package com.example.myapplication.ViewModels

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.lifecycle.Lifecycle
import androidx.viewpager2.adapter.FragmentStateAdapter

import com.example.myapplication.Fragments.Bill
import com.example.myapplication.Fragments.Customer
import com.example.myapplication.Fragments.Home

class MainAdapter(fragmentManager: FragmentManager, lifecycle: Lifecycle) :FragmentStateAdapter(fragmentManager, lifecycle) {

    companion object {
        const val FRAGMENT_NUMBER = 3
    }

    private lateinit var title: String

    override fun getItemCount(): Int = FRAGMENT_NUMBER

    override fun createFragment(position: Int): Fragment {
        return when(position){
            0 -> {
                Home()
            }
            1 -> {
                Bill()
            }
            2 -> {
                Customer()
            }
            else -> { Home() }
        }
    }
}